src/CBLURLEndpointListener_CAPI.cc

```diff
@@ -0,0 +1,85 @@
+//
+// CBLURLEndoiubtListeber_CAPI.cc
+//
+// Copyright © 2025 Couchbase. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "CBLURLEndpointListener_Internal.hh"
+
+#ifdef COUCHBASE_ENTERPRISE
+
+CBLListenerAuthenticator* CBLListenerAuth_CreatePassword(CBLListenerPasswordAuthCallback auth) noexcept {
+    return nullptr;
+}
+
+CBLListenerAuthenticator* CBLListenerAuth_CreateCertificate(CBLListenerCertAuthCallback auth) noexcept {
+    return nullptr;
+}
+
+void CBLListenerAuth_Free(CBLListenerAuthenticator* _cbl_nullable auth) noexcept {
+    delete auth;
+}
+
+CBLURLEndpointListener* CBLURLEndpointListener_Create(const CBLURLEndpointListenerConfiguration* conf, CBLError* outError) noexcept {
+    if (conf->collectionCount == 0) {
+        if (outError) {
+            outError->domain = kCBLDomain;
+            outError->code = kCBLErrorInvalidParameter;
+        }
+        return nullptr;
+    }
```

Comment:
To be inline with the implementation of the other API, I'd like to suggest moving the validation to the `CBLURLEndpointListener` class it self. Basically the constructor of the `CBLURLEndpointListener` class can validate and through InvalidParameter exception. This would make the glue code very thin.

Reply:
will do